linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -0,0 +1,258 @@
+# threadpool_linux
+
+## Overview
+
+`threadpool_linux` is the Linux-specific implementation of the `threadpool` module.
+
+## Design
+
+`threadpool_linux` provides the Linux implementation of `threadpool`. The `threadpool` object contains an array of threads all initialized in `threadpool_open_async`  and an array of tasks all initialized in `threadpool_create`, all incoming tasks are being added in `threadpool_schedule_work` and scheduled in `threadpool_work_func`. 
```

Comment:
> and an array of tasks all initialized in `threadpool_create` [](http://example.com/codeflow?start=166&length=61)

This is a bit confusing the way it is stated. I'd suggest breaking it out in a separate sentence like "The threadpool unit tracks the work items that need to be executed in an array." - does not even need to be mentioned at this level in which API the array is constructed.

Reply:
modified